### PR TITLE
add odinmini keyboard

### PR DESCRIPTION
**Pull request by moyi4681**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/odinmini/info.json [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        "on_state": 1
+    },
+    "matrix_pins": {
+        "cols": ["GP28", "GP26", "GP25", "GP24", "GP23", "GP22", "GP21", "GP20", "GP19", "GP18", "GP17", "GP16", "GP0", "GP7", "GP6", "GP5", "GP3", "GP15"],
+        "rows": ["GP1", "GP2", "GP29", "GP13", "GP14"]
+    },
+    "processor": "RP2040",
+    "rgblight": {
+        "animations": {
+        "all": true
```
Comment (drashna) [inline keyboards/kbdfans/odinmini/info.json:30]: Could you explicitly define which animations are enabled? 
```suggestion
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/odinmini/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		        ),	
+	[2]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
+		KC_TAB,  KC_Q,    KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_END,   KC_P7,   KC_P8,   KC_P9,   KC_PPLS, 
+		KC_CAPS, KC_A,    KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                      KC_P4,   KC_P5,   KC_P6,
+		KC_LSFT, KC_Z,    KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                       KC_P1,   KC_P2,    KC_P3,  KC_PENT, 
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RGHT,           KC_P0,    KC_PDOT
+		        ),
+	[3]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/via/keymap.c:43]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS, 
```

File: keyboards/kbdfans/odinmini/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                        ),
+	[1]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
+		KC_TAB,  KC_Q,    KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_END,   KC_P7,   KC_P8,   KC_P9,   KC_PPLS, 
+		KC_CAPS, KC_A,    KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                      KC_P4,   KC_P5,   KC_P6,
+		KC_LSFT, KC_Z,    KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                       KC_P1,   KC_P2,    KC_P3,  KC_PENT, 
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RGHT,           KC_P0,    KC_PDOT
+		        ),	
+	[2]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/via/keymap.c:36]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS, 
```

File: keyboards/kbdfans/odinmini/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+	[0] =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
+		KC_TAB,  KC_Q,    KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_END,   KC_P7,   KC_P8,   KC_P9,   KC_PPLS, 
+		KC_CAPS, KC_A,    KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                      KC_P4,   KC_P5,   KC_P6,
+		KC_LSFT, KC_Z,    KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                       KC_P1,   KC_P2,    KC_P3,  KC_PENT, 
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RGHT,           KC_P0,    KC_PDOT
+                        ),
+	[1]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/via/keymap.c:29]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS, 
```

File: keyboards/kbdfans/odinmini/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/via/keymap.c:22]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS, 
```

File: keyboards/kbdfans/odinmini/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+		        ),
+	[2]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+		KC_TAB,  KC_Q,    KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_END,   KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+		KC_CAPS, KC_A,    KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                      KC_P4,   KC_P5,   KC_P6,
+		KC_LSFT, KC_Z,    KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                       KC_P1,   KC_P2,    KC_P3,  KC_PENT,
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RGHT,           KC_P0,    KC_PDOT
+		        ),
+	[3]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/default/keymap.c:43]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
```

File: keyboards/kbdfans/odinmini/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+                        ),
+	[1]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+		KC_TAB,  KC_Q,    KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_END,   KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+		KC_CAPS, KC_A,    KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                      KC_P4,   KC_P5,   KC_P6,
+		KC_LSFT, KC_Z,    KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                       KC_P1,   KC_P2,    KC_P3,  KC_PENT,
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RGHT,           KC_P0,    KC_PDOT
+		        ),
+	[2]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/default/keymap.c:36]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
```

File: keyboards/kbdfans/odinmini/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+	[0] =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+		KC_TAB,  KC_Q,    KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_END,   KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+		KC_CAPS, KC_A,    KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                      KC_P4,   KC_P5,   KC_P6,
+		KC_LSFT, KC_Z,    KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                       KC_P1,   KC_P2,    KC_P3,  KC_PENT,
+		KC_LCTL, KC_LGUI, KC_LALT,          KC_SPC,           KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RGHT,           KC_P0,    KC_PDOT
+                        ),
+	[1]   =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/default/keymap.c:29]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
```

File: keyboards/kbdfans/odinmini/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] =LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/keymaps/default/keymap.c:22]:
```suggestion
		KC_ESC,  KC_1,    KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME,  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]
LGTM.

Though, for the layout macro, generally, `LAYOUT` if there is only one, and `LAYOUT_all` if there are multiple layouts.

File: keyboards/kbdfans/odinmini/info.json [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        "brightness_steps": 8,
+        "hue_steps": 8,
+        "led_count": 4,
+        "max_brightness": 180,
+        "pin": "GP12",
+        "saturation_steps": 8,
+        "sleep": true
+    },
+    "usb": {
+        "device_ver": "0x0001",
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/info.json:51]:
```suggestion
```

File: keyboards/kbdfans/odinmini/info.json [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        "command": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true
+    },
+    "indicators": {
+        "caps_lock": "GP27",
+        "num_lock": "GP4",
+        "on_state": 1
```
Comment (fauxpark) [inline keyboards/kbdfans/odinmini/info.json:21]:
```suggestion
        "num_lock": "GP4"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**